File: examples/src/networkInterfacesEnumeratorCSharp/networkInterfacesEnumerator.cs
Console.WriteLine("Using " + la_networkInterfaceHelper.getLibraryName() + " v" + la_networkInterfaceHelper.getLibraryVersion());
Console.WriteLine(la_networkInterfaceHelper.getLibraryCopyright());

displayInterfaces();

void displayInterfaces()
{
	Console.WriteLine("Available interfaces:");

	var obs = new Observer();
	la.networkInterface.NetworkInterfaceHelper.getInstance().registerObserver(obs);
	la.networkInterface.NetworkInterfaceHelper.getInstance().unregisterObserver(obs);
}

class Observer : la.networkInterface.NetworkInterfaceHelper.DefaultedObserver
{
	public override void onInterfaceAdded(la.networkInterface.Interface intfc)
	{
		Console.WriteLine(intNum + ": " + intfc.id);
		Console.WriteLine("  Description:  " + intfc.description);
		Console.WriteLine("  Alias:        " + intfc.alias);
		Console.WriteLine("  MacAddress:   " + intfc.macAddress);
		Console.WriteLine("  Type:         " + intfc.type);
		Console.WriteLine("  Enabled:      " + (intfc.isEnabled ? "YES" : "NO"));
		Console.WriteLine("  Connected:    " + (intfc.isConnected ? "YES" : "NO"));
		Console.WriteLine("  Virtual:      " + (intfc.isVirtual ? "YES" : "NO"));
		if (!intfc.ipAddressInfos.IsEmpty)
		{
			Console.WriteLine("  IP Addresses:");
			foreach (var info in intfc.ipAddressInfos)
			{
				if (info.address.getType() == la.networkInterface.IPAddress.Type.V4)
				{
					Console.WriteLine("    " + info.address.ToString() + " (" + info.netmask.ToString() + ") -> " + info.getNetworkBaseAddress().ToString() + " / " + info.getBroadcastAddress().ToString());
				}
				else
				{
					// IPv6
					Console.WriteLine("    " + info.address.ToString() + "/" + la.networkInterface.IPAddress.prefixLengthFromPackedV6(info.netmask.getIPV6Packed()) + " -> " + info.getNetworkBaseAddress().ToString());
				}
			}
		}
		if (!intfc.gateways.IsEmpty)
		{
			Console.WriteLine("  Gateways:");
			foreach (var ip in intfc.gateways)
			{
				Console.WriteLine("    " + ip.ToString());
			}
		}

		++intNum;
	}

	private uint intNum = 1;
};

File: examples/src/networkInterfacesEnumeratorLua/networkInterfacesEnumerator.lua
#!/usr/bin/env lua

-- Network Interfaces Enumerator Lua Example

-- Load the networkInterfaceHelper module
local nih = require("la_networkInterfaceHelper")
print("Using " .. nih.getLibraryName() .. " v" .. nih.getLibraryVersion())
print(nih.getLibraryCopyright())

function displayInterfaces()
	print("Available interfaces:")

	local intNum = 1
	local obs = nih.la.networkInterface.NetworkInterfaceHelper.DefaultedObserver()
	swig_override(obs, "onInterfaceAdded", function(self, intfc)
		print(intNum .. ": " .. intfc.id)
		print("  Description:  " .. intfc.description)
		print("  Alias:        " .. intfc.alias)
		print("  MacAddress:   " .. nih.la.networkInterface.NetworkInterfaceHelper.macAddressToString(intfc.macAddress))
		print("  Type:         " .. intfc.type)
		print("  Enabled:      " .. (intfc.isEnabled and "YES" or "NO"))
		print("  Connected:    " .. (intfc.isConnected and "YES" or "NO"))
		print("  Virtual:      " .. (intfc.isVirtual and "YES" or "NO"))

		if intfc.ipAddressInfos:size() > 0 then
			print("  IP Addresses:")
			for i = 0, intfc.ipAddressInfos:size() - 1 do
				local info = intfc.ipAddressInfos[i]
				if info.address:getType() == nih.la.networkInterface.IPAddress.Type_V4 then
					print("    " ..
						info.address:toString() ..
						" (" ..
						info.netmask:toString() ..
						") -> " .. info:getNetworkBaseAddress():toString() .. " / " .. info:getBroadcastAddress():toString())
				else
					-- IPv6
					print("    " ..
						info.address:toString() ..
						"/" ..
						nih.la.networkInterface.IPAddress.prefixLengthFromPackedV6(info.netmask:getIPV6Packed()) ..
						" -> " .. info:getNetworkBaseAddress():toString())
				end
			end
		end

		if intfc.gateways:size() > 0 then
			print("  Gateways:")
			for i = 0, intfc.gateways:size() - 1 do
				local ip = intfc.gateways[i]
				print("    " .. ip:toString())
			end
		end

		intNum = intNum + 1
	end)

	local helper = nih.la.networkInterface.NetworkInterfaceHelper.getInstance()
	helper:registerObserver(obs)
	helper:unregisterObserver(obs)
end

displayInterfaces()
